apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/servlet/ClientIpUtils.java

```diff
@@ -0,0 +1,53 @@
+package co.elastic.apm.servlet;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Utility class which helps to determine the real IP of a {@link HttpServletRequest}
+ * <p>
+ * This implementation is based on
+ * <a href="https://github.com/stagemonitor/stagemonitor/blob/0.88.2/stagemonitor-web-servlet/src/main/java/org/stagemonitor/web/servlet/MonitoredHttpRequest.java">stagemonitor</a>
+ * </p>
+ */
+public class ClientIpUtils {
+
+    /**
+     * This method returns the first IP from common HTTP header names used by reverse proxies.
+     * If there is no such header, it returns {@link HttpServletRequest#getRemoteAddr()}
+     *
+     * @param request The HTTP request.
+     * @return the
+     */
+    public static String getRealIp(HttpServletRequest request) {
```

Comment:
for reference, this is the apm-server implementation: https://github.com/elastic/apm-server/blob/12d38ced19a8a4a9555435e960bc47da4f824c6f/utility/ip.go#L12

Reply:
As the apm server is doing this anyway, I'll close this PR.